apps/system/js/operator_icon.js

```diff
@@ -0,0 +1,67 @@
+/* global BaseIcon, MobileOperator */
+'use strict';
+
+(function(exports) {
+  var OperatorIcon = function(manager) {
+    BaseIcon.call(this, manager);
+  };
+  OperatorIcon.prototype = Object.create(BaseIcon.prototype);
+  OperatorIcon.prototype.name = 'OperatorIcon';
+  OperatorIcon.prototype.update = function(now) {
+    if (!this.element || !this.enabled()) {
+      return;
+    }
+    this.show();
+    var _ = navigator.mozL10n.get;
+    var f = new navigator.mozL10n.DateTimeFormat();
+    now = now || new Date();
+    var l10nArgs = JSON.parse(this.element.dataset.l10nArgs || '{}');
+    l10nArgs.date = f.localeFormat(now, _('statusbarDateFormat'));
+    this.element.dataset.l10nArgs = JSON.stringify(l10nArgs);
+    var conns = this.manager.mobileConnections;
+    var conn;
+
+    // Do not show carrier's name if there are multiple sim cards.
+    if (conns && conns.length == 1) {
+      conn = conns[0];
+    }
+
+    var label = this.element;
+    var previousLabelContent = label.textContent;
+
+    if (!conn || !conn.voice || !conn.voice.connected ||
+        conn.voice.emergencyCallsOnly) {
+      delete l10nArgs.operator;
+      label.dataset.l10nArgs = JSON.stringify(l10nArgs);
+
+      label.dataset.l10nId = '';
+      label.textContent = l10nArgs.date;
+
+
+      this.debug(previousLabelContent, label.textContent);
+      if (previousLabelContent !== label.textContent) {
+        this.publish('widthchanged');
+      }
+
+      return;
+    }
+
+    var operatorInfos = MobileOperator.userFacingInfo(conn);
+    l10nArgs.operator = operatorInfos.operator;
+
+    if (operatorInfos.region) {
+      l10nArgs.operator += ' ' + operatorInfos.region;
+    }
+
+    label.dataset.l10nArgs = JSON.stringify(l10nArgs);
+
+    label.dataset.l10nId = 'statusbarLabel';
+    label.textContent = _('statusbarLabel', l10nArgs);
```

Comment:
this line is not needed. and the above two you can replace with:
navigator.mozL10n.setAttributes(label, l10nId, l10nArgs); :)